eng/Versions.props

```diff
@@ -44,7 +44,7 @@
     -->
     <MicrosoftCodeAnalysisCSharpVersion>4.10.0-2.24114.13</MicrosoftCodeAnalysisCSharpVersion>
     <MicrosoftCodeAnalysisVersion>4.10.0-2.24114.13</MicrosoftCodeAnalysisVersion>
-    <MicrosoftNetCompilersToolsetVersion>4.10.0-2.24114.13</MicrosoftNetCompilersToolsetVersion>
+    <MicrosoftNetCompilersToolsetVersion>4.10.0-3.24162.9</MicrosoftNetCompilersToolsetVersion>
```

Comment:
@ViktorHofer, I forget what the rules are about bumping this, in particular with regards to source build. Is it ok?

Reply:
Not sure. @dotnet/source-build-internal

Reply:
Yeah, so this is causing a prebuilt: `Microsoft.Net.Compilers.Toolset.4.10.0-3.24162.9`. What I don't get is why this wasn't causing a prebuilt before with the older version. First, nothing in the binlog shows this package being installed, so it's not done via the normal restore means. I assumed that the previous version was working because it aligned with the runtime repo's SDK version from global.json which is set to preview.1. But that SDK version is associated with `4.10.0-1.24067.21`, not `4.10.0-2.24114.13`. So how was this working before?

/cc @MichaelSimons

Reply:
This package comes from roslyn which builds after runtime in the product source-build.  This means when building in source-build the N-1 version will be used during the build.  The reason the version declared here is not used is because there is a corresponding entry in the Versions.Details.xml.  That is the signal to source-build to lift the version to current/n-1 depending on the build dependency order.  This is described in https://github.com/dotnet/source-build/blob/main/Documentation/package-dependency-flow.md.

This all means that in order to not break source-build you should not take a dependency newer than the version just released yesterday.  Also to get your repo level source-leg to pass, you are going to need to update the Version.Details.xml roslyn dependencies.

Reply:
> This all means that in order to not break source-build you should not take a dependency newer than the version just released yesterday.

Thanks. Just to make sure I'm understanding, it's fine to update this / Version.Details.xml to whatever the next to most recent build is? e.g. if the most recently available build is 19 it's fine to set this to 18?

Reply:
> it's fine to update this / Version.Details.xml to whatever the next to most recent build is

Not quite - it is appropriate to set it to what the most recent released version is - what ever was released yesterday as part of 9.0 preview 2.  If you need a newer version, that will require distro maintainers to rebootstrap during the 9.0-preview 3 release because they can't use their preview 2 build to build preview 3.  We try to limit requiring rebootstraps during previews.  Rebootstraps are not allowed during servicing and RC releases.

Reply:
Ah, ok. Where do I find what compiler toolset version maps to 9.0 preview 2? (And for future reference where I can find the version corresponding to preview 3, 4, etc.)

Reply:
This is an easy way: https://github.com/dotnet/dotnet/blob/v9.0.0-preview.2.24128.5/prereqs/git-info/roslyn.props#L6. We produce tags for each release that can be used (in this case, `v9.0.0-preview.2.24128.5`).

Reply:
Thanks!